File: java/leetcode_grind/Day698.java
package leetcode_grind;

import java.util.*;

public class Day698 {
    // https://leetcode.com/problems/maximal-score-after-applying-k-operations/description/?envType=daily-question&envId=2024-10-14
    static class Solution1 {
        public long maxKelements(int[] nums, int k) {
            var pq = new PriorityQueue<Integer>((a, b) -> b - a);
            for (var num : nums) {
                pq.add(num);
            }
            var ans = 0l;
            while (k-- > 0) {
                var p = pq.poll();
                ans += p;
                pq.add((int) Math.ceil((double) p / 3.0));
            }
            return ans;
        }
    }
}

File: src/leetcode_grind/day_698.rs
// https://leetcode.com/problems/maximal-score-after-applying-k-operations/description/?envType=daily-question&envId=2024-10-14
pub fn max_kelements(nums: Vec<i32>, mut k: i32) -> i64 {
    use std::collections::BinaryHeap;
    let mut pq = BinaryHeap::new();
    for num in nums {
        pq.push(num);
    }
    let mut ans = 0;
    while k > 0 {
        let top = pq.pop().unwrap();
        ans += top as i64;
        pq.push(((top as f64) / 3.0).ceil() as i32);
        k -= 1;
    }
    ans
}
